### PR TITLE
Dat 3091 - refactoring

### PR DESCRIPTION
**Pull request by nikodamn**
Refactoring Portable Infobox tests

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java [hunk 1]
```diff
@@ -119,8 +117,11 @@ public void verifyCategoriesInTemplateInvocation() {
     SourceEditModePageObject
         src =
         info.navigateToArticleEditPageSrc(wikiURL, PageContent.PI_TEMPLATE_WEBSITE_SIMPLE);
+    System.out.println("1");
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)

File: src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java [hunk 2]
```diff
@@ -119,8 +117,11 @@ public void verifyCategoriesInTemplateInvocation() {
...
     src.focusTextArea();
+    System.out.println("2");
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)

File: src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java [hunk 3]
```diff
@@ -119,8 +117,11 @@ public void verifyCategoriesInTemplateInvocation() {
...
     String catName = src.getRandomDigits(9);
+    System.out.println("3");
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/category/CategoryPageObject.java [hunk 4]
```diff
@@ -28,6 +28,7 @@ public String getCategoryName() {
 
     public void verifyCategoryPageTitle(String title) {
         wait.forElementVisible(categoryHeader);
+        System.out.println("Category Header: " + categoryHeader.getText());
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/category/CategoryPageObject.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 5 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/eda1de2f268ec440a1add94aa17b478ef51237ea/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfoboxPageObject.java#L90) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
* MINOR [Reorder the modifiers to comply with the Java Language Specification.](https://github.com/Wikia/selenium-tests/blob/eda1de2f268ec440a1add94aa17b478ef51237ea/src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java#L50) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by nikodamn** [comment]
@tomasznapieralski :)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/dc9d66eb78799465caa87646fd3d8a39377b0385/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfoboxPageObject.java#L90) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
* MINOR [Reorder the modifiers to comply with the Java Language Specification.](https://github.com/Wikia/selenium-tests/blob/dc9d66eb78799465caa87646fd3d8a39377b0385/src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java#L50) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfoboxPageObject.java [hunk 5]
```diff
@@ -14,6 +17,7 @@
  * Created by Rodriuki on 11/06/15.
  * Created by nikodamn on 20/07/15
  */
+// TODO: Clean up
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfoboxPageObject.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java [hunk 6]
```diff
@@ -111,7 +113,8 @@ public void verifyImagesInWhatLinksHerePage() {
     links.verifyInfoboxArticleInList(articleName);
   }
 
-  @Test(groups = {"PortableInfoboxTests", "PortableInfoboxTests_004"})
+  // TODO: compress this test
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java [hunk 7]
```diff
@@ -130,83 +133,70 @@ public void verifyCategoriesInTemplateInvocation() {
...
-    WebElement horizontalItemValue = info.getHorizontalItemValue();
-    WebElement itemValue = info.getItemValue();
-    info.compareFontSizes(horizontalItemValue, itemValue);
+    new PortableInfoboxPageObject(driver)
+        .open(PageContent.PORTABLE_INFOBOX01)
+        .compareFontSizesBetweenHorizontalItemLabelAndItemLabel()
+        .compareFontSizesBetweenHorizontalItemValueAndItemValue();
   }
 
+  // TODO: compress this test
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java [hunk 8]
```diff
@@ -62,20 +64,19 @@ public void verifyElementsVisibility() {
     info.verifyReferencesPresence();
     info.verifyImagePresence();
     info.verifyInfoboxTitlePresence();
-
   }
 
-  @Test(groups = {"PortableInfoboxTests", "PortableInfoboxTests_002"})
+  // TODO: compress this test
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java [hunk 9]
```diff
@@ -46,15 +45,18 @@
  * TC18: Additional <div> wrappers from title, header and image HTML are removed
  * TC19: Verify that any of the tags which do not have a value won't appear
  */
+@Test(groups = "PortableInfoboxTests")
 public class PortableInfoboxTests extends NewTestTemplate {
 
   private final static Logger LOGGER = Logger.getLogger(PortableInfoboxTests.class.getName());
 
-  @Test(groups = {"PortableInfoboxTests", "PortableInfoboxTests_001"})
+  // TODO: compress this test
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java [hunk 10]
```diff
@@ -99,10 +100,11 @@ public void verifyElementsRedirects() {
     info.compareURLAndInternalLink(internalLinkName, internalNavigatedURL);
   }
 
-  @Test(groups = {"PortableInfoboxTests", "PortableInfoboxTests_003"})
+  // TODO: compress this test
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java [hunk 11]
```diff
@@ -130,83 +133,70 @@ public void verifyCategoriesInTemplateInvocation() {
...
-    PortableInfoboxPageObject info = article.getInfoboxPage();
-    info.verifyTabberPresence();
-    info.verifyTabberImagePresence();
+    new PortableInfoboxPageObject(driver)
+        .open(PageContent.PORTABLE_INFOBOX02)
+        .verifyTabberPresence()
+        .verifyTabberImagePresence();
   }
 
+  // TODO: compress this test
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/category/CategoryPageObject.java [hunk 12]
```diff
@@ -10,25 +10,24 @@
 /**
  * Created by nikodem on 27.07.15.
  */
+// TODO: What is this class about?
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/category/CategoryPageObject.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 8 issues:
* INFO 8 info

Watch the comments in this conversation to review them.

**Comment by tomasznapieralski** [comment]
more tomorrow

File: src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java [hunk 13]
```diff
@@ -4,6 +4,7 @@
 import com.wikia.webdriver.common.core.Assertion;
 import com.wikia.webdriver.common.core.annotations.Execute;
 import com.wikia.webdriver.common.core.annotations.User;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/portableinfoboxtests/PortableInfoboxTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.logging.PageObjectLogging'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfoboxPageObject.java [hunk 14]
```diff
@@ -1,8 +1,13 @@
 package com.wikia.webdriver.pageobjectsfactory.pageobject;
 
+import com.wikia.webdriver.common.contentpatterns.PortableInfobox;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfoboxPageObject.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.contentpatterns.PortableInfobox'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfoboxPageObject.java [hunk 15]
```diff
@@ -220,36 +243,133 @@ public CreateArticleModalComponentObject clickRedLink(int i) {
...
+    return this;
+  }
+
+  public PortableInfoboxPageObject areBoldElementsMoreThanOne() {
+    Assertion.assertTrue(boldElements.size() > 0);
+    return this;
+  }
+
+  public PortableInfoboxPageObject areItalicElementsMoreThanOne() {
+    Assertion.assertTrue(italicElements.size() > 0);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfoboxPageObject.java]: MAJOR Use isEmpty() to check whether the collection is empty or not. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1155)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfoboxPageObject.java [hunk 16]
```diff
@@ -220,36 +243,133 @@ public CreateArticleModalComponentObject clickRedLink(int i) {
...
+    return this;
+  }
+
+  public PortableInfoboxPageObject areHeadersMoreThanOne() {
+    Assertion.assertTrue(pInfoTitleH3.size() > 0);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfoboxPageObject.java]: MAJOR Use isEmpty() to check whether the collection is empty or not. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1155)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfoboxPageObject.java [hunk 17]
```diff
@@ -220,36 +243,133 @@ public CreateArticleModalComponentObject clickRedLink(int i) {
...
+  }
+
+  public PortableInfoboxPageObject open(String articleTitle) {
+    getUrl(urlBuilder.getUrlForWiki(Configuration.getWikiName()) + URLsContent.WIKI_DIR
+           + articleTitle);
+    return this;
+  }
+
+  public PortableInfoboxPageObject areBoldElementsMoreThanOne() {
+    Assertion.assertTrue(boldElements.size() > 0);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfoboxPageObject.java]: MAJOR Use isEmpty() to check whether the collection is empty or not. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1155)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 10 issues:
* MAJOR 4 major
* MINOR 6 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Define a constant instead of duplicating this literal "selected " 3 times.](https://github.com/Wikia/selenium-tests/blob/9af1bd090b2116733b63de6d47925a45bd43d2b9/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L251) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Immediately return this expression instead of assigning it to the temporary variable "tablePropertiesContent".](https://github.com/Wikia/selenium-tests/blob/9af1bd090b2116733b63de6d47925a45bd43d2b9/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L232) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)
* MAJOR [Remove this unused "j" local variable.](https://github.com/Wikia/selenium-tests/blob/9af1bd090b2116733b63de6d47925a45bd43d2b9/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L289) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)
* MINOR [Define a constant instead of duplicating this literal "addComponent" 4 times.](https://github.com/Wikia/selenium-tests/blob/9af1bd090b2116733b63de6d47925a45bd43d2b9/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L251) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal " component" 3 times.](https://github.com/Wikia/selenium-tests/blob/9af1bd090b2116733b63de6d47925a45bd43d2b9/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L251) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.pageobject.article.ArticlePageObject'.](https://github.com/Wikia/selenium-tests/blob/9af1bd090b2116733b63de6d47925a45bd43d2b9/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/template/TemplatePageObject.java#L5) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by nikodamn** [comment]
@tomasznapieralski thanks!

**Comment by WikiaTech** [comment]
SonarQube analysis reported 14 issues:
* MAJOR 6 major
* MINOR 8 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Add a default case to this switch.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L858) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
* MAJOR [Add a default case to this switch.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L818) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
* MINOR [Define a constant instead of duplicating this literal "DropdownVerified" 3 times.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L393) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Add a default case to this switch.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L798) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
* MINOR [Replace this "switch" statement by "if" statements to increase readability.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L895) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1301)
* MAJOR [Make this final field static too.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L169) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1170)
* MAJOR [Add a default case to this switch.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L838) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
* MINOR [Define a constant instead of duplicating this literal "selected " 3 times.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L248) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Remove this unused "j" local variable.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L286) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)
* MINOR [Define a constant instead of duplicating this literal "addComponent" 4 times.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L248) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Remove this unused import 'org.openqa.selenium.support.PageFactory'.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L24) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MINOR [Define a constant instead of duplicating this literal " component" 3 times.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L248) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Immediately return this expression instead of assigning it to the temporary variable "tablePropertiesContent".](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L229) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)
* MINOR [Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.pageobject.article.ArticlePageObject'.](https://github.com/Wikia/selenium-tests/blob/bda57900a872e3a2ae3280a214fb3cc4d9743ffa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/template/TemplatePageObject.java#L5) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues:
* MAJOR 1 major
* MINOR 6 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Immediately return this expression instead of assigning it to the temporary variable "tablePropertiesContent".](https://github.com/Wikia/selenium-tests/blob/dd5c539140f9bbb57596d1a02afa295f33442c84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L229) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)
* MINOR [Define a constant instead of duplicating this literal "selected " 3 times.](https://github.com/Wikia/selenium-tests/blob/dd5c539140f9bbb57596d1a02afa295f33442c84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L248) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "addComponent" 4 times.](https://github.com/Wikia/selenium-tests/blob/dd5c539140f9bbb57596d1a02afa295f33442c84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L248) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal " component" 3 times.](https://github.com/Wikia/selenium-tests/blob/dd5c539140f9bbb57596d1a02afa295f33442c84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L248) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Remove this unused import 'org.openqa.selenium.support.PageFactory'.](https://github.com/Wikia/selenium-tests/blob/dd5c539140f9bbb57596d1a02afa295f33442c84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L24) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MAJOR [Remove this unused "j" local variable.](https://github.com/Wikia/selenium-tests/blob/dd5c539140f9bbb57596d1a02afa295f33442c84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L286) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)
* MINOR [Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.pageobject.article.ArticlePageObject'.](https://github.com/Wikia/selenium-tests/blob/dd5c539140f9bbb57596d1a02afa295f33442c84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/template/TemplatePageObject.java#L5) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by nikodamn** [comment]
@tomasznapieralski should be better now. ;)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 14 issues:
* MAJOR 6 major
* MINOR 8 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Add a default case to this switch.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L858) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
* MINOR [Define a constant instead of duplicating this literal "DropdownVerified" 3 times.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L393) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Add a default case to this switch.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L818) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
* MINOR [Replace this "switch" statement by "if" statements to increase readability.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L895) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1301)
* MAJOR [Make this final field static too.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L169) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1170)
* MAJOR [Add a default case to this switch.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L798) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
* MAJOR [Add a default case to this switch.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java#L838) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
* MINOR [Define a constant instead of duplicating this literal "addComponent" 4 times.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L248) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal " component" 3 times.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L248) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "selected " 3 times.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L248) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Remove this unused import 'org.openqa.selenium.support.PageFactory'.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L24) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MINOR [Immediately return this expression instead of assigning it to the temporary variable "tablePropertiesContent".](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L229) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)
* MAJOR [Remove this unused "j" local variable.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/SourceEditModePageObject.java#L286) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)
* MINOR [Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.pageobject.article.ArticlePageObject'.](https://github.com/Wikia/selenium-tests/blob/0a633be0efc1a63269f94e07c180f98d6ed9b243/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/template/TemplatePageObject.java#L5) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by tomasznapieralski** [comment]
LGTM

**Pull request merged**